New Debian issues after merging ROS 1 stuff

- [x] Build in i386 is broken due to failing unit tests, apparently in the conversion of gridmap values between ROS and MRPT maps: [log](https://launchpadlibrarian.net/438118104/buildlog_ubuntu-eoan-i386.mrpt_1%3A1.9.9~snapshot20190820-0956-git-eb3744a2-eoan-1_BUILDING.txt.gz)

```
[ RUN      ] Map.check_ros2mrpt_and_back
/<<PKGBUILDDIR>>/libs/ros1bridge/src/map_unittest.cpp:92: Failure
The difference between 1.0f - 0.01f * i and desMrpt.getCell(i, 0) is 0.99964302935404703, which exceeds 0.03f, where
1.0f - 0.01f * i evaluates to 1,
desMrpt.getCell(i, 0) evaluates to 0.00035697064595296979, and
0.03f evaluates to 0.029999999329447746.
ros to mprti=0
/<<PKGBUILDDIR>>/libs/ros1bridge/src/map_unittest.cpp:95: Failure
The difference between srcRos.data[i] and desRos.data[i] is 100, which exceeds 1, where
srcRos.data[i] evaluates to 0,
desRos.data[i] evaluates to 100, and
1 evaluates to 1.
ros to mprt to ros
[  FAILED  ] Map.check_ros2mrpt_and_back (3 ms)
```

After a quick inspection, I didn't see any obvious source of this problem... postponing. (Anyone?)

- [x] Xenial packages are broken due to a bug in vtk6 / PCL packaging. See:
https://github.com/PointCloudLibrary/pcl/issues/1594#issuecomment-283873617
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=819741

Our error is: [log](https://launchpad.net/~joseluisblancoc/+archive/ubuntu/mrpt/+build/17443160)
```
/usr/bin/ld: cannot find -lvtkproj4
```
(vtkproj4 is not a real lib, but something internal to vtk)


@jolting : Didn't you have a VTK PPA for Xenial with a more modern version of vtk? I could add it as a PPA dependency to MRPT2 PPA.

PS: Good news is... `mrpt-ros1bridge` and the app `rosbag2rawlog` are already available from the PPA, and will be available from the official Ubuntu repos when mrpt 2 gets released :tada:

--- a/libs/maps/include/mrpt/maps/CLogOddsGridMapLUT.h
+++ b/libs/maps/include/mrpt/maps/CLogOddsGridMapLUT.h
@@ -10,6 +10,7 @@
 #pragma once
 
 #include <mrpt/maps/logoddscell_traits.h>
+#include <cmath>
 
 namespace mrpt::maps
 {
@@ -56,7 +57,7 @@ struct CLogOddsGridMapLUT : public detail::logoddscell_traits<TCELL>
 		logoddsTable_255.resize(traits_t::LOGODDS_LUT_ENTRIES);
 		for (int i = traits_t::CELLTYPE_MIN; i <= traits_t::CELLTYPE_MAX; i++)
 		{
-			float f = 1.0f / (1.0f + exp(-i * LOGODD_K_INV));
+			float f = 1.0f / (1.0f + std::exp(-i * LOGODD_K_INV));
 			unsigned int idx = -traits_t::CELLTYPE_MIN + i;
 			logoddsTable[idx] = f;
 			logoddsTable_255[idx] = (uint8_t)(f * 255.0f);
@@ -64,16 +65,11 @@ struct CLogOddsGridMapLUT : public detail::logoddscell_traits<TCELL>
 
 		// Build the p2lTable as well:
 		p2lTable.resize(traits_t::P2LTABLE_SIZE + 1);
-		double K = 1.0 / traits_t::P2LTABLE_SIZE;
+		const double K = 1.0 / traits_t::P2LTABLE_SIZE;
 		for (int j = 0; j <= traits_t::P2LTABLE_SIZE; j++)
 		{
-			double p = j * K;
-			if (p == 0)
-				p = 1e-14;
-			else if (p == 1)
-				p = 1 - 1e-14;
-
-			double logodd = log(p) - log(1 - p);
+			const double p = std::min(1.0 - 1e-14, std::max(1e-14, j * K));
+			const double logodd = log(p) - log(1 - p);
 			int L = round(logodd * LOGODD_K);
 			if (L > traits_t::CELLTYPE_MAX)
 				L = traits_t::CELLTYPE_MAX;

--- a/libs/maps/src/maps/CLogOddsGridMapLUT_unittest.cpp
+++ b/libs/maps/src/maps/CLogOddsGridMapLUT_unittest.cpp
@@ -1,0 +1,50 @@
+/* +------------------------------------------------------------------------+
+   |                     Mobile Robot Programming Toolkit (MRPT)            |
+   |                          https://www.mrpt.org/                         |
+   |                                                                        |
+   | Copyright (c) 2005-2019, Individual contributors, see AUTHORS file     |
+   | See: https://www.mrpt.org/Authors - All rights reserved.               |
+   | Released under BSD License. See: https://www.mrpt.org/License          |
+   +------------------------------------------------------------------------+ */
+
+#include <gtest/gtest.h>
+#include <mrpt/maps/CLogOddsGridMapLUT.h>
+#include <numeric>
+
+template <typename cell_t>
+void test_monotonic()
+{
+    mrpt::maps::CLogOddsGridMapLUT<cell_t> lut;
+    float last_p=.0f;
+    const auto i_init = std::numeric_limits<cell_t>::min();
+    const auto i_end = std::numeric_limits<cell_t>::max();
+    for (int64_t i=i_init;i<=i_end;i++)
+    {
+        // Ensure LUT is monotonic:
+        const float new_p = lut.l2p(i);
+        EXPECT_GE(new_p, last_p) << " i=" << i;
+        last_p = new_p;
+	}
+	EXPECT_NEAR(lut.l2p(i_init), .0f, 0.05);
+	EXPECT_NEAR(lut.l2p(i_end), 1.0f, 0.05);
+
+	// Expect internal table to be monotonic:
+	int64_t last_logodd = std::numeric_limits<int64_t>::min();
+	for (size_t idx = 0; idx < lut.p2lTable.size(); idx++)
+	{
+		const int64_t next_logodd = lut.p2lTable[idx];
+		EXPECT_GE(next_logodd, last_logodd) << "idx=" << idx;
+
+		last_logodd = next_logodd;
+	}
+}
+
+TEST(CLogOddsGridMapLUT, monotonic_8bit)
+{
+    test_monotonic<int8_t>();
+}
+
+TEST(CLogOddsGridMapLUT, monotonic_16bit)
+{
+    test_monotonic<int16_t>();
+}

--- a/libs/ros1bridge/include/mrpt/ros1bridge/map.h
+++ b/libs/ros1bridge/include/mrpt/ros1bridge/map.h
@@ -31,11 +31,11 @@ class MapHdl
 {
    private:
 #ifdef OCCUPANCY_GRIDMAP_CELL_SIZE_8BITS
-	int8_t lut_cellmrpt2ros[0x100];  // lookup table for entry convertion
+	int lut_cellmrpt2ros[0x100];  // lookup table for entry convertion
 #else
-	int8_t lut_cellmrpt2ros[0xFFFF];  // lookup table for entry convertion
+	int lut_cellmrpt2ros[0xFFFF];  // lookup table for entry convertion
 #endif
-	int8_t lut_cellros2mrpt[101];  // lookup table for entry convertion
+	int lut_cellros2mrpt[101];  // lookup table for entry convertion
 
 	MapHdl();
 	MapHdl(const MapHdl&);

